libraries/AP_LandingGear/AP_LandingGear.cpp

```diff
@@ -286,18 +281,17 @@ void AP_LandingGear::update(float height_above_ground_m)
 
     if (hal.util->get_soft_armed()) {
         // only do height based triggering when armed
-        if ((!_deployed || !_have_changed) &&
+        if (!_deployed  &&
             _deploy_alt > 0 &&
             alt_m <= _deploy_alt &&
             _last_height_above_ground > _deploy_alt) {
             deploy();
-        }
-        if ((_deployed || !_have_changed) &&
-            _retract_alt > 0 &&
-            _retract_alt >= _deploy_alt &&
-            alt_m >= _retract_alt &&
-            _last_height_above_ground < _retract_alt) {
-            retract();
+        } else if (_deployed  &&
```

Comment:
this will break the case of DEPLOY_ALT=0 and RETRACT_ALT>0, and no aux function used, it will no longer retract on first takeoff

Reply:
it does in SITL...servo 5 is the retract servo
LGR_DEPLOY_ALT   0.000000
LGR_DEPLOY_PIN   -1.000000 # Disabled
LGR_DEPLOY_POL   0.000000 # Low
LGR_ENABLE       1.000000 # Enabled
LGR_OPTIONS      2.000000 # Deploy during Land
LGR_RETRACT_ALT  20.000000
LGR_STARTUP      2.000000 # Deploy
LGR_WOW_PIN      -1.000000 # Disabled
LGR_WOW_POL      0.000000 # Low

![gear](https://github.com/user-attachments/assets/a36760c1-3df1-4bff-a878-3072df60c703)
Also @IamPete1 concerns about it not retracting again on a touch and go don't occur

Reply:
tthe only way behaviour diverges from master if LGR_STARTUP = 0 and the pilot does NOT deploy the gear before  setting it on the runway for takeoff via MAVLink with no AUX switch....but this is almost a certain crash, since the landing gear servo will be getting 0 PWM and have no torque...the 0 case , as Iunderstand it, was to prevent landing gear from changing position on the bench when powered up unexpectedly...even if the gear is down, it will potentially collapse on takeoff roll or sooner if the gear servo is not getting PWM...so this is a non-issue in my opinion...

Reply:
added back have_changed